text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1003 @@
+# Proposal
+[proposal]: #proposal
+
+Cargo configuration will contain a new key `build-std` under the `[build]`
+section ([?][rationale-build-std-in-config]), permitting one of two values -
+"off" ([?][rationale-build-std-off]) or "always", defaulting to "off":
+
+```toml
+[build]
+build-std = "always" # or `off`
+```
+
+`build-std` can also be specified in the `[target.<triple>]` and
+`[target.<cfg>]` sections ([?][rationale-build-std-target-section]):
+
+```toml
+[target.aarch64-unknown-illumos]
+build-std = "always" # or `off`
+```
+
+The `build-std` configuration locations have the following precedence
+([?][rationale-build-std-precedence]):
+
+1. `[target.<triple>]`
+2. `[target.<cfg>]`
+3. `[build]`
+
+As the Cargo configuration is local to the current installation of Cargo
+(typically in `~/.config/cargo`), the value of `build-std` is not influenced by
+the dependencies of the current crate.
+
+When `build-std` is set to "always", then the standard library will be
+unconditionally recompiled ([?][rationale-unconditional]) in its release profile
+as part of every clean build ([?][rationale-release-profile]). This is primarily
+useful for users of tier three targets.
```

Comment:
Whose definition of the release profile?  `std`s or the users?

Reply:
`std`'s - the idea being that we're building something as close to the pre-built std as we can, and therefore limiting how much new surface area of the standard library we're potentially exposing (i.e. configurations of it that we're not used to supporting), at least initially